pkg/spoke/spokeagent.go

```diff
@@ -177,7 +193,8 @@ func (o *SpokeAgentOptions) RunSpokeAgent(ctx context.Context, controllerContext
 		clientCertForHubController := managedcluster.NewClientCertForHubController(
 			o.ClusterName, o.AgentName, o.ComponentNamespace, o.HubKubeconfigSecret,
 			kubeconfigData,
-			spokeKubeClient.CoreV1(),
+			// store the secret in the cluster where the agent pod runs
```

Comment:
I think we should add some more comment here. Maybe also a TODO to revisit whether we should put secret for addon on agent cluster or spoke cluster.